arex-instrumentation/config/arex-apollo/src/main/java/io/arex/inst/config/apollo/ApolloConfigExtractor.java

```diff
@@ -0,0 +1,90 @@
+package io.arex.inst.config.apollo;
+
+import io.arex.agent.bootstrap.model.Mocker;
+import io.arex.agent.bootstrap.util.StringUtil;
+import io.arex.inst.runtime.log.LogManager;
+import io.arex.inst.runtime.serializer.Serializer;
+import io.arex.inst.runtime.util.MockUtils;
+
+import java.util.Properties;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+
+public class ApolloConfigExtractor {
+    private static final AtomicBoolean UPDATED = new AtomicBoolean(true);
+    private static final AtomicReference<String> RECORD_CONFIG_BATCH_NO = new AtomicReference<>(null);
+    private static final long ONE_MINUTE_EXPIRED_NANOS_TIME = TimeUnit.MINUTES.toNanos(1);
+    private static boolean startReplay = false;
+    private static long replayStartTime = 0;
+    private static String currentReplayConfigBatchNo;
+
+    public static ApolloConfigExtractor tryCreateExtractor() {
+        if (UPDATED.compareAndSet(true, false)) {
+            RECORD_CONFIG_BATCH_NO.set(UUID.randomUUID().toString());
+            return new ApolloConfigExtractor();
+        }
+        return null;
+    }
+    public void record(String fileName, Properties properties) {
+        Mocker mocker = MockUtils.createConfigFile(fileName);
+        mocker.setRecordId(RECORD_CONFIG_BATCH_NO.get());
+        mocker.setRecordEnvironment(1);
+        mocker.getTargetResponse().setBody(Serializer.serialize(properties));
+        mocker.getTargetResponse().setType(Properties.class.getName());
```

Comment:
Is it better to set it as a static field

Reply:
got it, fixed